File: frontend/src/api/revision.ts
import apiClient from './client'

export interface RevisionComment {
  id: string
  order_id: string
  user_id: string
  comment: string
  revision_number: number
  created_at: string
  user_name?: string
  user_email?: string
}

export const getRevisionComments = async (orderId: string): Promise<RevisionComment[]> => {
  try {
    const response = await apiClient.get(`/orders/${orderId}/revision-comments`)
    return response.data
  } catch (error: any) {
    console.error('🔍 Get revision comments error:', error.response?.data)
    throw new Error(error.response?.data?.detail || 'Ошибка загрузки комментариев')
  }
}

File: backend/app/schemas/revision.py
# app/schemas/revision.py
"""
Схемы для комментариев правок
"""
from datetime import datetime
from typing import Optional
from uuid import UUID
from pydantic import BaseModel, Field

class RevisionCommentBase(BaseModel):
    comment: str = Field(..., min_length=1, max_length=2000)

class RevisionCommentCreate(RevisionCommentBase):
    order_id: UUID

class RevisionComment(RevisionCommentBase):
    id: UUID
    order_id: UUID
    user_id: UUID
    revision_number: int
    created_at: datetime
    
    # Связи
    user: Optional[dict] = None
    
    class Config:
        from_attributes = True

class RevisionCommentWithUser(RevisionComment):
    """Схема с информацией о пользователе"""
    user_name: Optional[str] = None
    user_email: Optional[str] = None
    
    class Config:
        from_attributes = True